File: Kotlin/tests/BoardTest.kt
class BoardTest() {
    fun run() {
        canCheckForBlackCell()
        canCheckForWhiteCell()
        canFlipBlackCell()
        canFlipWhiteCell()
    }

    fun canCheckForBlackCell(){
        var board = Board(array(Position(1,1)).toSet())
        assert(board.isBlack(Position(1,1)))
    }

    fun canCheckForWhiteCell(){
        var board = Board(array(Position(1,1)).toSet())
        assert(!board.isBlack(Position(2,2)))
    }

    fun canFlipBlackCell() {
        var board = Board(array(Position(1,1)).toSet())
        board.flip(Position(1,1))
        assert(!board.isBlack(Position(1,1)))
    }

    fun canFlipWhiteCell() {
        var board = Board(array(Position(2,2)).toSet())
        board.flip(Position(1,1))
        assert(board.isBlack(Position(1,1)))
    }
}

File: Dart/test/BoardTest.dart
class BoardTest {
  run() {
    group('Board.isBlack', () {
      
      var blackCell = new Cell(0, 0);
      var whiteCell = new Cell(1, 0);
      var board = new Board([blackCell]);
      
      test('return false for white cell', () {
        Expect.isFalse(board.isBlack(whiteCell));
      });
      
      test('returns true for black cell', () {
        Expect.isTrue(board.isBlack(blackCell));
      });
    });
    
    group('Board.flipColor', () {
      
      var blackCell = new Cell(0, 0);
      var whiteCell = new Cell(1, 0);
      
      test('can flip white cell to black', () {
        Board board = new Board([]);
        board.flipColor(whiteCell);
        Expect.isTrue(board.isBlack(whiteCell));
      });
      
      test('can flip black cell to white', () {
        Board board = new Board([blackCell]);
        board.flipColor(blackCell);
        Expect.isFalse(board.isBlack(blackCell));
      });
    });
  }
}
